### ISSUE TITLE
Terraform support for Origin Groups in azurerm_cdn_endpoint

### ISSUE THREAD

**Issue by skn200** [hashicorp/terraform-provider-azurerm#10771]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Terraform support for Origin Groups in azurerm_cdn_endpoint . Not able to create Origin Groups in azurerm_cdn_endpoint. As per azure arm template origin group is an independent block, but iam not able to find any Origin Group block in the this document [azurerm_cdn_endpoint] (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cdn_endpoint)

Is there any other way to use Origin Groups

### New or Affected Resource(s)

azurerm_cdn_endpoint

**Comment by jjnp** [comment]
Also, there is now the option to use `Origin Group Override` as an Action in the Rules Engine. This too should be added to Terraform to keep it up to date with Azure's feature set.

**Comment by Jaykul** [comment]
Currently, CDNs created with "Storage" backend (instead of storage static website) don't work properly without this, because you have to set the **origin host header** to make them work (which, as far as I can tell, **requires** creating a default _origin group_).

**Comment by samiulseequent** [comment]
Is there any progress on this issue?

**Comment by ownerer** [comment]
Bump, any updates here?

**Comment by ryepup** [comment]
Would love to see this implemented. My workaround is create my CDN using `azurerm_cdn_profile`, and then create my endpoints using `azurerm_resource_group_template_deployment` and ARM templates.

**Comment by didayal-msft** [comment]
I am working on building the azure verified module for cdn profile and checking if there any update on this one?

**Comment by WodansSon** [comment]
This work is blocked by an upstream Microsoft swagger issue. To implement this ask we need API Version:
2024-02-01. The current provider API version is 2020-09-01.

**Comment by david-garcia-garcia** [comment]
The upstream issue that prevented this from being done seems to have been fixed int the azure API:

https://github.com/Azure/azure-rest-api-specs/pull/29629

This is the PR @WodansSon was working on before being dropped due to the upstream issue:

https://github.com/hashicorp/terraform-provider-azurerm/pull/25744/

**Comment by davemooreuws** [comment]
Hey @WodansSon , 

Since the upstream issue has been fixed. Are you planning on reopening your PR? Thanks

**Comment by WodansSon** [comment]
I have added this work item back to my radar for investigation.

**Comment by davemooreuws** [comment]
> I have added this work item back to my radar for investigation.

Given the deprecation of front door classic, this might not be needed anymore. We just moved to front door standard.

https://azure.microsoft.com/en-au/updates?id=azure-front-door-classic-will-be-retired-on-31-march-2027

**Comment by WodansSon** [comment]
@davemooreuws, that is correct. I have looked into supporting this and have received a `no-go` since the resource(s) are being deprecated by the Azure Service Team.  You can see a consolidated view of the reasoning [here](https://gist.github.com/WodansSon/4045137c99c77ff29f93a3084d71efcb). Microsoft as a whole is strongly suggesting that current users of the Classic CDN resource(s) move to the Azure CDN Front Door resource(s) instead, which are fully supported by the Terraform provider.
